admin/services/pupil-register-v2.service.js

```diff
@@ -22,7 +41,10 @@ const pupilRegisterV2Service = {
         group: d.groupName
       }
     })
-    return pupilIdentificationFlagService.addIdentificationFlags(tableSorting.applySorting(pupilRegister, 'lastName'))
+    const pupilRegisterViewData = pupilIdentificationFlagService.addIdentificationFlags(tableSorting.applySorting(pupilRegister, 'lastName'))
+    const pupilRegisterRedisKey = `school:${schoolId}`
+    await redisCacheService.set(pupilRegisterRedisKey, pupilRegisterViewData)
```

Comment:
you should check if the data exists in the cache, and return it.  if it doesnt exist, get from sql, add to cache and return

Reply:
@GuyHarwood This is done by the `getPupilRegister` which calls `getPupilRegisterViewData` once it identifies that no data exist in the cache.
If `getPupilRegister` detects data in cache `getPupilRegisterViewData` won't be called at all.